modules/Bio/EnsEMBL/Compara/Utils/Cigars.pm

```diff
@@ -100,14 +102,19 @@ sub compose_sequence_with_cigar {
     throw("Invalid cigar_line '$cigar_line'\n") if $cigar_line !~ /^[0-9A-Z]*$/;
```

Comment:
OK, but with a question mark instead of the second `*` ?
```suggestion
     throw("Invalid cigar_line '$cigar_line'\n") if $cigar_line !~ /^(([1-9][0-9]*)?[A-Z])*$/;
```